### ISSUE TITLE
Deprecate (remove) "no label" modifiers 

### ISSUE THREAD

**Issue by nonprofittechy** [SuffolkLITLab/docassemble-AssemblyLine#565]
"No label" doesn't play nicely with screen readers. We should rephrase screens so that every question has a label

**Comment by BryceStevenWilley** [comment]
Limited use case, but what works slightly better than no label is if it's the only field on the page is a dropdown, directly using a `choices` dropdown.

I do remember running into this and attempting an upstream fix for pages with just one field, but I don't remember if I gave it up for the above fix instead. There wasn't a great fix for `no label` though, but there just isn't as much information from the author, so I changed it to duplicate information already found on the page (and already read out by the screen reader), just associating it directly with the field.

**Comment by nonprofittechy** [comment]
@BryceStevenWilley are you saying that something like this:

```yaml
question: |
  Is anyone else on your side of the case?
field: users.there_is_another
choices:
  - Yes: True
  - No: False
```

Will be read correctly by the screen reader?

**Comment by BryceStevenWilley** [comment]
I did end up making the upstream change, but it was only for dropdowns. Previously was invalid according to accessibility checkers, fixed by: https://github.com/jhpyle/docassemble/commit/384b82f06507bbbdd2a62cafee6c5358915da78f, which explicitly adds `aria-labelledby` to the select option. 

I'm not sure if the suggestion applies to `yesnoradio`s though. The HTML of `choices` vs `fields`in that case is pretty similar.

Form controls in screen readers are still a thing that I don't fully understand, so I'm hesitant to say that `no label` is okay. Reading through top to bottom is better than not being able to see content, but most screen reader users aren't going to do that for each screen, they'll skip around and find the context they need (it's also a bit harder to tell how long the page will be at the beginning, so I assume that skipping around before realizing the short length of the page is likely). My lack of knowledge is how screen readers announce and handle jumping around. The automated accessibility checkers seem to like the radio buttons okay, but the button group still doesn't have an explicit label (in any of those cases! I'll try to make another DA fix)

**Comment by nonprofittechy** [comment]
I'm still confused about the problems and correct solutions here. 

Maybe we can add to agenda for next Monday?

**Comment by BryceStevenWilley** [comment]
The problem is that each of these screens doesn't have the correct ARIA to label the group of radio buttons. From [MDN docs on the "radiogroup" role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/roles/radiogroup_role):

> The radiogroup must have an accessible name either by a visible label referenced by [aria-labelledby](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby) or has a label specified with [aria-label](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label). If elements provide additional information about the radio group, those elements are referenced by the radiogroup element with the [aria-describedby](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby) property.

This should have a detrimental effect on screen reader usage, but I don't know enough about advanced screen reader usage to show exactly what the issue is. 

IMO the correct solution here is to improve upstream. All of these cases could have the right information for the ARIA for button groups on screen (wouldn't need extra info the interview author), but need DA fixes to correctly link the group to the label (either the author given label, or the page header).

I won't be present for this upcoming Monday, but the week after would be fine I guess?

**Comment by nonprofittechy** [comment]
Just discussed on admin call, we think that the idea of automatically adding the aria-labelledby is correct.

But we also want an `aria-label` modifier upstream, if possible, so people can override the label for a specific field.

**Comment by nonprofittechy** [comment]
That's fine, the upstream label helps

**Comment by nonprofittechy** [comment]
I think we've actually solved all of these. We have a comprehensive way to catch this issue now with DAYamlChecker that knows the nuances.
